include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -3361,6 +3361,61 @@ namespace internal
   }
 
 
+  template <int dim, int n_points_1d_template, typename Number>
+  inline bool
+  compute_weights_fe_q_dofs_by_entity(const Number *     data,
+                                      const unsigned int n_components,
+                                      const int n_points_1d_non_template,
+                                      Number *  weights)
+  {
+    const int n_points_1d = n_points_1d_template != -1 ?
+                              n_points_1d_template :
+                              n_points_1d_non_template;
+
+    Assert(n_points_1d > 0, ExcNotImplemented());
+    Assert(n_points_1d < 100, ExcNotImplemented());
+
+    unsigned int compressed_index[100];
+    compressed_index[0] = 0;
+    for (int i = 1; i < n_points_1d - 1; ++i)
+      compressed_index[i] = 1;
+    compressed_index[n_points_1d - 1] = 2;
+
+    // Helper function that extracts the weight. Before doing so,
+    // it checks the values already set.
+    const auto set = [](Number &weight, const Number &data) {
```

Comment:
Can we call this function `check_and_set`, and extend the comment to say something like "Insert the number `data` into a storage position for `weight`, ensuring that the array has either not been touched before or the previous content is the same. In case the previous content has a different value, we exit this function and signal to outer functions that the compression was not possible."